src/hooks/useFetchEvent.ts

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from "react";
+import { type ModifiedResult } from "~/pages/api/events/all";
+
+const useFetchEvent = (id: string | null, date: Date | null) => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [fetchedData, setFetchedData] = useState<
+    ModifiedResult[] | ModifiedResult | null
+  >(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setLoading(true);
+        if (id) {
+          const response = await fetch(`/api/events/${id}`);
+          const data = (await response.json()) as ModifiedResult | null;
+          setFetchedData(data);
+        } else if (date) {
+          const response = await fetch("/api/events/all", {
+            method: "POST",
+            body: JSON.stringify({ date }),
+          });
+          const data = (await response.json()) as ModifiedResult[];
+          setFetchedData(data);
+        }
+
+        setLoading(false);
+      } catch (error) {
+        let message = "Unknown Error";
+        if (error instanceof Error) message = error.message;
+        setError(message);
+        setLoading(false);
+      }
+    };
+
+    void fetchData();
+  }, [id, date]);
```

Comment:
@Lada496 In L37, I'm just curious about the usage of `void` word just before `fetchData()`. Correct me if I'm wrong but this `void` implies that `fetchData` function returns nothing, doesn't it?

(And is it common to add `void` keyword like this in the professional settings?)

Reply:
@keento0809 
not sure if it's the same case, but I also used `void` when fetching data in useEffect.
It gives a _typescript-eslint error_ without `void`
Here is what I referred to: https://stackoverflow.com/questions/43980188/what-could-this-be-about-tslint-error-promises-must-be-handled-appropriately

👇 Error 👇 
<img width="748" alt="" src="https://github.com/van-squad/map-t3/assets/68415905/a36c91cd-5bb1-479f-9b0b-8a80fac9c8ab">

Reply:
@devjoylee Oh, thank you for sharing the informative stuff! Now I totally understand the usage of `void` here!